kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/internal/parquet/ParquetFileReaderSuite.scala

```diff
@@ -0,0 +1,276 @@
+    goldenTableFile("parquet-decimal-dictionaries-v2"))
+
+  test("decimals encoded using dictionary encoding ") {
+    val expResult = (0 until 1000000).map { i =>
+      TestRow(i, BigDecimal.valueOf(i%5), BigDecimal.valueOf(i%6), BigDecimal.valueOf(i%2))
+    }
+
+    val readSchema = new StructType()
+      .add("id", IntegerType.INTEGER)
+      .add("col1", new DecimalType(9, 0)) // INT32: 1 <= precision <= 9
+      .add("col2", new DecimalType(12, 0)) // INT64: 10 <= precision <= 18
+      .add("col3", new DecimalType(25, 0)) // FIXED_LEN_BYTE_ARRAY
+
+    for (file <- Seq(DECIMAL_TYPES_DICT_FILE_V1, DECIMAL_TYPES_DICT_FILE_V2)) {
+      val actResult = readParquetFilesUsingSpark(file, readSchema)
+
+      checkAnswer(actResult, expResult)
+    }
+  }
+
+  private val LARGE_SCALE_DECIMAL_TYPES_FILE = getSingleParquetFile(
+    goldenTableFile("parquet-decimal-type"))
+
+  test("large scale decimal type file") {
+
+    def expand(n: BigDecimal): BigDecimal = {
+      n.scaleByPowerOfTen(5).add(n)
+    }
+
+    val expResult = (0 until 99998).map { i =>
+      if (i % 85 == 0) {
+        val n = BigDecimal.valueOf(i)
+        TestRow(i, n.movePointLeft(1).setScale(1), n.setScale(5), n.setScale(5))
+      } else {
+        val negation = if (i % 33 == 0) {
+          -1
+        } else {
+          1
+        }
+        val n = BigDecimal.valueOf(i*negation)
+        TestRow(
+          i,
+          n.movePointLeft(1),
+          expand(n).movePointLeft(5),
+          expand(expand(expand(n))).movePointLeft(5)
+        )
+      }
+    }
+
+    val readSchema = new StructType()
+      .add("id", IntegerType.INTEGER)
+      .add("col1", new DecimalType(5, 1)) // INT32: 1 <= precision <= 9
+      .add("col2", new DecimalType(10, 5)) // INT64: 10 <= precision <= 18
+      .add("col3", new DecimalType(20, 5)) // FIXED_LEN_BYTE_ARRAY
+
+    val actResult = readParquetFilesUsingSpark(LARGE_SCALE_DECIMAL_TYPES_FILE, readSchema)
+
+    checkAnswer(actResult, expResult)
+  }
+
+  private val ALL_TYPES_FILE = getSingleParquetFile(goldenTableFile("parquet-all-types"))
+
+  test("read all types of data") {
+    val readSchema = tableSchema(goldenTablePath("parquet-all-types"))
+
+    val actResult = readParquetFilesUsingSpark(ALL_TYPES_FILE, readSchema)
+
+    val expResult = generateRowsFromAllTypesFile(readSchema)
+
+    checkAnswer(actResult, expResult)
+  }
+
+  test("read subset of columns") {
+    val readSchema = new StructType()
+      .add("byteType", ByteType.BYTE)
+      .add("booleanType", BooleanType.BOOLEAN)
+      .add("stringType", StringType.STRING)
+      .add("dateType", DateType.DATE)
+      .add("nested_struct", new StructType()
+        .add("aa", StringType.STRING)
+        .add("ac", new StructType().add("aca", IntegerType.INTEGER)))
+      .add("array_of_prims", new ArrayType(IntegerType.INTEGER, true))
+
+    val actResult = readParquetFilesUsingSpark(ALL_TYPES_FILE, readSchema)
```

Comment:
actual results should be ready by Kernel reader. Use `readParquetFilesUsingKernel`. 

For the expected results use `readParquetFilesUsingSpark`, so that we can remove the methods that currently hardcoded the expected results: `generateRowsFromAllTypesFile

Reply:
Same comment for few other tests.

Reply:
> actual results should be ready by Kernel reader. Use `readParquetFilesUsingKernel`.
>
@vkorukanti got it. but, could you explain the reason for me pls. this part i just copy old code from `TestParquetBatchReader` to here.

> For the expected results use `readParquetFilesUsingSpark`, so that we can remove the methods that currently hardcoded the expected results: `generateRowsFromAllTypesFile
>
i'm sorry, i don't understand this one, `readParquetFilesUsingSpark` from what?

Reply:
In the current tests (before this PR):

1) Generate the actual results using the Kernel Parquet reader.
2) Generate the expected results using `generateRowsFromAllTypesFile` which basically hardcoded the expected results.
3) compare the actual and expected results

We want to change this to:
1) Generate the actual results using the Kernel Parquet reader. This step is same as before, but use the new utility method `readParquetFilesUsingKernel` that directly returns the `Seq[TestRow]`
2) Generate the expected results using `readParquetFilesUsingSpark(parquet file directory)`. This step is different. Instead of using the hardcoded expected results, we are relying on the spark to read the Parquet files and get the contents.
3) compare the actual and expected results. This step is same as before.


`val inputLocation = goldenTablePath("parquet-all-types")` - this gives the directory of Parquet file contents. Pass this as input to `readParquetFilesUsingSpark`.

Reply:
Oh, got it. Thank you so much!